jax/experimental/checkify/checkify_impl.py

```diff
@@ -82,19 +89,24 @@ def throw(self):
     if err:
       raise ValueError(err)
 
+
 register_pytree_node(Error,
-                     lambda e: ((e.err, e.code), tuple(sorted(e.msgs.items()))),
-                     lambda msgs, data: Error(*data, dict(msgs)))  # type: ignore
+                     lambda e: ((e.err, e.code, e.payload),
+                                tuple(sorted(e.msgs.items()))),
+                     lambda msgs, data: Error(data[0], data[1],  # type: ignore
+                                              dict(msgs), data[2]))  # type: ignore
 
 init_error = Error(False, 0, {})
 next_code = it.count(1).__next__  # globally unique ids, could be uuid4
 
 
-def assert_func(error: Error, pred: Bool, msg: str) -> Error:
+def assert_func(error: Error, pred: Bool, msg: str,
+                payload: Payload = np.ones((), np.int32)) -> Error:
```

Comment:
```suggestion
                payload: Payload) -> Error:
```

Do we need a default arg here? Defaults are dangerous and this is an explicit-ish function anyway, so maybe we can drop it.

Reply:
Nevermind, Lena explained to me why it's better to make this optional.